scripts/india_census/primary_religion_data/preprocess.py

```diff
@@ -0,0 +1,354 @@
+            name_array.append("Female")
+
+        name = "_".join(name_array)
+        row["name"] = name
+
+        key = "{0}_{1}".format(
+            row["columnName"],
+            place_of_residence if place_of_residence != None else "Total")
+
+        if data_category:
+            key = key + "_" + titleCase(data_category)
+
+        self.stat_var_index[key] = name
+        row["StatisticalVariable"] = self._get_stat_var_name(name)
+        row["measuredProperty"] = self._get_measured_property_name(name)
+        self.mcf.append(row)
+
+        stat_var = self.GENERIC_TEMPLATE_STAT_VAR.format(**dict(row))
+
+        return name, stat_var
+
+    def _create_mcf(self):
+        self.mcf = []
+        with open(self.metadata_file_path) as csvfile:
+            reader = csv.DictReader(csvfile)
+            with open(self.mcf_file_path, 'w+', newline='') as f_out:
+                for data_row in reader:
+                    for place_of_residence in [None, "Urban", "Rural"]:
+                        if len(self.data_categories) == 0:
+                            name, stat_var = self._create_variable(
+                                data_row,
+                                place_of_residence,
+                                data_category=None)
+                            #if the statvar already exists then we don't
+                            #need to recreate it
+                            if name in self.existing_stat_var:
+                                pass
+                            #we need to create statvars only for those columns that
+                            #exist in the current data file
+                            elif data_row[
+                                    "columnName"] not in self.census_columns:
+                                pass
+                            else:
+                                f_out.write(stat_var)
+                        else:
+                            for data_category in self.data_categories:
+                                name, stat_var = self._create_variable(
+                                    data_row,
+                                    place_of_residence,
+                                    data_category=data_category)
+                                #if the statvar already exists then we don't
+                                #need to recreate it
+                                if name in self.existing_stat_var:
+                                    pass
+                                #we need to create statvars only for those columns that
+                                #exist in the current data file
+                                elif data_row[
+                                        "columnName"] not in self.census_columns:
+                                    pass
+                                else:
+                                    f_out.write(stat_var)
+
+
+if __name__ == '__main__':
+
+    metadata_file_path = os.path.join(
```

Comment:
Doesn't primary_abstract_data_variables.csv define the schema? Why do we need it now?

Reply:
The  `Religion PCA data` has a different XLS format but has same columns for example
```
No_HH
TOT_P
TOT_M
TOT_F
..
..
```
Along with addition constraints.

So I am reusing `primary_abstract_data_variables.csv` just to derive meaningful names for the stat vars and nothing else.

Also when we plan to define schema. It would be very easy to do that.